pkg/server/proto/server.proto

```diff
@@ -5420,6 +5472,169 @@ message ProjectTemplate {
+}
+
+// ListAddOnDefinitionsResponse is the response used to return all Add On
+// definitions
+message ListAddOnDefinitionsResponse {
+  repeated AddOnDefinition add_on_definitions = 1;
+  PaginationResponse pagination = 2;
+  uint64 total_count = 3;
+}
+
+// CreateAddOnDefinitionRequest is the request used to create an Add On
+// definition
+message CreateAddOnDefinitionRequest {
+  AddOnDefinition add_on_definition = 1;
+}
+
+// CreateAddOnDefinitionResponse is the response containing the just-created
+// Add On definition
+message CreateAddOnDefinitionResponse {
+  AddOnDefinition add_on_definition = 1;
+}
+
+// UpdateAddOnDefinitionRequest is the request used to update an existing
+// Add On definition
+message UpdateAddOnDefinitionRequest {
+  AddOnDefinition add_on_definition = 1;
+}
+
+// UpdateAddOnDefinitionResponse is the response containing the just-updated
+// Add On definition
+message UpdateAddOnDefinitionResponse {
+  AddOnDefinition add_on_definition = 1;
+}
+
+// DeleteAddOnDefinitionRequest is the request used to delete an existing Add
+// On definition
+message DeleteAddOnDefinitionRequest {
+  Ref.AddOnDefinition add_on_definition = 1;
+}
+
+// AddAddOnRequest is the request used to add an Add On to a project
+message AddAddOnRequest {
+  AddOn add_on = 1;
+}
+
+// AddAddOnResponse is the response containing the just-added Add On
+message AddAddOnResponse {
+  AddOn add_on = 1;
+}
+
+// DestroyAddOnRequest is the request used to destroy an existing Add On
+message DestroyAddOnRequest {
+  Ref.AddOn add_on = 1;
+}
+
+// GetAddOnRequest is the request used to get an existing Add On
+message GetAddOnRequest {
+  Ref.AddOn add_on = 1;
+}
+
+// GetAddOnResponse is the response containing the just-requested existing Add
+// On
+message GetAddOnResponse {
+  AddOn add_on = 1;
+}
+
+// ListAddOnsRequest is the request used to get all existing Add Ons
+message ListAddOnsRequest {
+  PaginationRequest pagination = 1;
+}
+
+// ListAddOnsResponse is the response containing all existing Add Ons
+message ListAddOnsResponse {
+  repeated AddOnDefinition definitions = 1;
+  PaginationResponse pagination = 2;
+}
+
+// GetTFRunStatusRequest is the request used to get the status of the latest
+// run of a Terraform workspace
+message GetTFRunStatusRequest {
+  string workspace_name = 1;
+}
+
+// GetTFRunStatusResponse is the response containing the status of the latest
+// run of a Terraform workspace
+message GetTFRunStatusResponse {
+  // the URL of the TF run
+  string url = 1;
+
+  // the state of the TF run
+  State state = 2;
+
+  enum State {
+    UNKNOWN = 0;
+    RUNNING = 1;
+    SUCCESS = 2;
+    ERROR   = 3;
+  }
+}
+
+// AddOn is an add-on definition rendered for a project
+message AddOn {
+  string id = 1;
+  string name = 2;
+  Ref.Project project = 3;
+  Ref.AddOnDefinition definition = 4;
+  string tf_workspace_name = 5;
+  bytes readme_markdown = 6;
+  repeated string tags = 7;
+  Type type = 8;
```

Comment:
Add variable doc strings here on these please! It's nice for code docs when interacting with protobuf messages.